lib/vanagon/component.rb

```diff
@@ -204,7 +204,7 @@ def get_source(workdir) # rubocop:disable Metrics/AbcSize
         extract_with << source.extract(platform.tar) if source.respond_to? :extract
 
         @cleanup_source = source.cleanup if source.respond_to?(:cleanup)
-        @dirname = source.dirname
+        @dirname ||= source.dirname
```

Comment:
The naive answer is "move it into its own getter":

```ruby
def dirname
  @dirname ||= source.dirname
end
```

But thinking that through, it looks like source initialization is wonky and brittle. I planned to unwind this soon for the `render` command so maybe I'll fast track that. Your way probably works for now but I think that a lot of `get_source()` is going to be broken up and spread around soon.